airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -826,6 +828,38 @@ def test_read(
                 detailed_logger=detailed_logger,
             )
 
+    def test_emitted_at_increase_on_subsequent_runs(
+        self,
+        connector_config,
+        configured_catalog,
+        docker_runner: ConnectorRunner,
+    ):
+        first_read_output = docker_runner.call_read(connector_config, configured_catalog)
+        first_read_records = filter_output(first_read_output, Type.RECORD)
+        first_read_records_data = [message.record.data for message in first_read_records]
+
+        assert first_read_records_data, "At least one record should be read using provided catalog"
+
+        first_read_records_emitted_at = [message.record.emitted_at for message in first_read_records]
+        max_emitted_at = max(first_read_records_emitted_at)
+
+        # sleep for 1 second to ensure that the emitted_at timestamp is different
+        time.sleep(1)
+
+        second_read_output = docker_runner.call_read(connector_config, configured_catalog)
+        second_read_records = filter_output(second_read_output, Type.RECORD)
+        second_read_records_data = [message.record.data for message in second_read_records]
+        second_read_reords_emitted_at = [message.record.emitted_at for message in second_read_records]
+
+        min_emitted_at = min(second_read_reords_emitted_at)
+
+        assert max_emitted_at < min_emitted_at, "emitted_at should increase on subsequent runs"
+
+        # Compare two lists, ignoring order of elements.
+        diff = DeepDiff(first_read_records_data, second_read_records_data, ignore_order=True, report_repetition=True)
+
+        assert diff == {}, f"records should be the same on subsequent runs. Diff: {diff}"
```

Comment:
I think this "two subsequent runs" logic is already happening [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_full_refresh.py#L39). I would suggest adding your new assertions in `TestFullRefresh.test_sequentials_reads` to avoid code duplication and limit the number of READ attempts. Let me know if you think otherwise.

Reply:
> limit the number of READ attempts

I think thats a good reason to change. Ill put up a commit shortly

Reply:
Actually @alafanechere .

How much of an impact do reads have to run time?

Should we always been trying to optomize for a low number of reads to run the SAT suite?

Reply:
It highly depends on the connector under test but its definitely one of the most time-consuming step of the test.

> Should we always been trying to optomize for a low number of reads to run the SAT suite?

I'd say we shall indeed optimize to lower the number of reads and implement caching whenever possible.